### PR TITLE
Add `snap_notify` documentation

### PR DESCRIPTION
**Pull request by GuillaumeRx**
Fixes https://github.com/MetaMask/snaps-skunkworks/issues/418

### CODE REVIEW

File: docs/guide/snaps-rpc-api.md [hunk 1]
```diff
@@ -596,3 +596,70 @@ await wallet.request({
...
+}
+```
+
+#### Returns
+
+`null` - This method doesn't return any data.
+
+#### Description
+
+Calling this method causes a notification to be added in the MetaMask UI.
```
Comment (FrederikBolding) [inline docs/guide/snaps-rpc-api.md:650]: Only `inApp` notifications are shown in the UI. Can we distinguish here?

Comment (FrederikBolding) [reply docs/guide/snaps-rpc-api.md:650]: "Calling this method shows a notification natively in the browser or in MetaMask" or something like that?

File: docs/guide/snaps-rpc-api.md [hunk 2]
```diff
@@ -596,3 +596,70 @@ await wallet.request({
...
+
+```typescript
+enum NotificationType {
+  /**
+   * A default browser notification
+   */
+  native = 'native',
+
+  /**
+   * A in-app notification stored in the metamask state.
```
Comment (FrederikBolding) [inline docs/guide/snaps-rpc-api.md:637]: I think "An in-app notification showed inside the MetaMask UI" might be better, remove the line below.

**Comment by FrederikBolding** [comment]
To be merged when Flask ships this commit: https://github.com/MetaMask/metamask-extension/commit/b599035a12010e31b85c3733fb573ba05b840b3c

**Review by FrederikBolding — DISMISSED** [review]

**Review by rekmarks — APPROVED** [review]

**Pull request merged**